src/alloc.rs

```diff
@@ -377,8 +377,10 @@ pub struct PtrInfo {
     pub gc: bool,
 }
 
-fn abort() {
-    unsafe { core::intrinsics::abort() };
+fn abort(msg: &str) {
```

Comment:
It might not be a good idea to have a function which matches a well-known builtin (`abort`) but with different semantics (this `abort` cleans up after itself). Maybe name it something like `alloc_exit`?

Reply:
Good idea